scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -1591,33 +1594,51 @@ def __handle_syslog_logs(self, data, extra):
                         limit_key="syslog-max-log-files",
                     )
                 else:
-                    logfile = AutoFlushingRotatingFile(
-                        filename=path,
-                        max_bytes=self.__max_log_size,
-                        backup_count=self.__max_log_rotations,
-                        flush_delay=self.__flush_delay,
-                    )
+                    # Match the new log path with a global log config (generally globbed).
+                    # Note that in log configs, .path is required but .attributes may be empty.
+                    log_config_attribs = None
+                    for log_config in self.__global_log_configs:
+                        if fnmatch(path, log_config["path"]):
+                            log_config_attribs = log_config["attributes"]
+                            break
+
+                    # No match logs to the base syslog file
+                    if log_config_attribs is not None:
+                        logfile = AutoFlushingRotatingFile(
+                            filename=path,
+                            max_bytes=self.__max_log_size,
+                            backup_count=self.__max_log_rotations,
+                            flush_delay=self.__flush_delay,
+                        )
 
-                    attribs = self.__syslog_attributes.copy()
-                    attribs.update(
-                        {k.lower(): v for k, v in extra.items() if v is not None}
-                    )
+                        attribs = self.__syslog_attributes.copy()
```

Comment:
Does this dictionary only contain simple values? In case it doesn't, we may need to use ``copy.deepcopy()`` to avoid surprises.

Reply:
For testing reasons, it's probably also a good idea to refactor some of the functionality here in separate standalone utility functions.

Reply:
Yup they're scalar values

Reply:
I think it could also aid with troubleshooting if add some debug log statements here as well (actual log config object, etc.).

Reply:
Sure, will do